public/js/components/Editor.js

```diff
@@ -71,45 +71,53 @@ const Editor = React.createClass({
     }
   },
 
-  clearDebugLine(line) {
-    this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
-  },
-
-  setDebugLine(line) {
-    this.editor.codeMirror.addLineClass(line - 1, "line", "debug-line");
-    this.editor.alignLine(line);
-  },
-
-  setSourceText(newSourceText, oldSourceText) {
-    if (newSourceText.get("loading")) {
-      this.setText("Loading...");
-      return;
+  updateDebugLine(prevProps, nextProps) {
+    if (prevProps.selectedFrame) {
+      const line = prevProps.selectedFrame.location.line;
+      this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
     }
-
-    if (newSourceText.get("error")) {
-      this.setText("Error");
-      console.error(newSourceText.get("error"));
-      return;
+    if (nextProps.selectedFrame) {
+      const line = nextProps.selectedFrame.location.line;
+      this.editor.codeMirror.addLineClass(line - 1, "line", "debug-line");
     }
+  },
 
-    this.setText(newSourceText.get("text"));
-    this.setMode(newSourceText);
-
-    resizeBreakpointGutter(this.editor.codeMirror);
+  highlightLine() {
+    // If the location has changed and a specific line is requested,
+    // move to that line and flash it.
+    if (this.pendingJumpLine) {
```

Comment:
this is a good candidate for a guard statement. It would more clearly say, don't highlight a line unless we're jumping

```js
if (!this.pendingJumpLine) {
  return;
}
```